scipy/optimize/_gensa.py

```diff
@@ -0,0 +1,607 @@
+##############################################################################
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
```

Comment:
The GPL licence is incompatible with Scipy, so needs to be removed. I think @sgubianpm is also the author of the GenSA R package. The [GPL FAQ](http://www.gnu.org/licenses/gpl-faq.en.html#TranslateCode) says: 
> Under copyright law, translation of a work is considered a kind of modification. Therefore, what the GPL says about modified versions applies also to translated versions.

I think this means that we need to know that this PR isn't a translation of the package. Given that @sgubianpm wrote both there are inevitably going to be similarities.